.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+name: ci
+on:
+  push:
+    branches: [main, rlamb/exponential-doubling-capacity-scaling]
+    paths-ignore:
+      - '**.md' #Do not need to run CI for markdown changes.
+  pull_request:
+    branches: [main]
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  build-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [14, latest]
+      max-parallel: 1
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+      - name: NPM Install
+        run: npm install
+      - name: Run Unit Tests
+        run: npm run test
+      - name: Build Contract Tests
+        run: make build-contract-tests
+      - name: Run Contract Tests
+        run: |
```

Comment:
@keelerm84 Does this whole way of running things make sense to you? Your shell-fu is better, it seems to all do what I want to get two nice logs and propagate errors correctly.